### ISSUE TITLE
document.getElementById

### ISSUE THREAD

**Issue by canda** [henrybuilt/react-sticky-table#16]
here you are doing 
https://github.com/henrybuilt/react-sticky-table/blob/master/src/index.js#L47

```
  componentDidMount() {
    this.table = document.getElementById('sticky-table-' + this.id);
```

why don't you use a ref instead? https://reactjs.org/docs/refs-and-the-dom.html

```
render() {
  [...]
    return (
      <div className={'sticky-table ' + (this.props.className || '')} ref={(tableWrapper) => this.table = tableWrapper}>

```

**Comment by maxhudson** [comment]
@canda good question. That'll have to be rolled up into v2

**Comment by canda** [comment]
Are you working on a v2 already? Which changes are coming?

**Comment by maxhudson** [comment]
We have a working v2 that works with jQuery, it just has to be ported to React. There is no explicit timeline on that yet, but I would like to execute soon.

- The implementation is much cleaner structurally
- Scrollbar appearance is improved
- Multiple sticky headers/columns can be specified
- There are no double-rendered cells behind the sticky columns/headers
- Vertical scrolling is entirely native now as opposed to being controlled by JS

**Comment by maxhudson** [comment]
@canda sticking away from this so we can refer to these elements outside componentDid<...> methods.

**Comment by canda** [comment]
@maxhudson 
But snapshot tests fail because of this random id.
Also it's not 100% safe to use this MATH.random (I know the chances are really minimal).
I think on react it's an expected behavior to have to wait for the lifecycle events to use references to the dom.

**Comment by maxhudson** [comment]
@canda agreed random id is far from perfect. I can allow an id to be specified as a middle-ground solution. Would that solve the issue you're having at the moment?

We are waiting for lifecycle events to use references, but then we need to use them in-between lifecycle events to resize the table on events like browser resize and cell resize. React refs are not available in-between lifecycle events as far as I know.

This package generally diverges from standard React patterns, which I'm not happy about, but was the best I could do.  I'm confident there is a more "Reactive" way to do this, but I'm not sure how. So far, not really getting help from others who know more.

**Comment by avilaj** [comment]
I'm also having troubles with this. 
Also it should be easier to solve using latest api from react 16.3 
`React.createRef();`

**Comment by maxhudson** [comment]
@avilaj can you please describe the issue you're having so I can determine if using `createRef()` will solve it? I'll likely end up implementing this anyway.

**Comment by avilaj** [comment]
@maxhudson my problem is with snapshot testing, they fail due to a random generated id on `react-sticky-tables`.
Following it a bit, I've seen that a reference might work there. But I understand that now might be hard to solve due to ref availability on the different lifeCycles. As I understood `createRef()` could be used even from constructor.

**Comment by maxhudson** [comment]
@avilaj great. I'll be implementing a fix this weekend.
